File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesAggregateCastCredit.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Aggregate cast credit for a TMDB TV series.
 *
 * @param adult              whether the person is marked as adult
 * @param gender             TMDB gender value
 * @param id                 TMDB person id
 * @param knownForDepartment known department
 * @param name               person name
 * @param originalName       original person name
 * @param popularity         person popularity
 * @param profilePath        profile image path
 * @param roles              aggregate roles
 * @param totalEpisodeCount  total episode count
 * @param order              cast order
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesAggregateCastCredit(
        boolean adult,

        int gender,

        int id,

        @JsonProperty("known_for_department")
        String knownForDepartment,

        String name,

        @JsonProperty("original_name")
        String originalName,

        double popularity,

        @JsonProperty("profile_path")
        String profilePath,

        List<TvSeriesAggregateCastRole> roles,

        @JsonProperty("total_episode_count")
        int totalEpisodeCount,

        int order
) implements TmdbModel {

    public TvSeriesAggregateCastCredit {
        roles = TmdbCollections.list(roles);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesAggregateCastCredit.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Aggregate cast credit for a TMDB TV series.
 *
 * @property adult whether the person is marked as adult
 * @property gender TMDB gender value
 * @property id TMDB person id
 * @property knownForDepartment known department
 * @property name person name
 * @property originalName original person name
 * @property popularity person popularity
 * @property profilePath profile image path
 * @property roles aggregate roles
 * @property totalEpisodeCount total episode count
 * @property order cast order
 */
@JvmRecord
data class TvSeriesAggregateCastCredit(
    @all:JsonProperty("adult")
    val adult: Boolean,

    @all:JsonProperty("gender")
    val gender: Int,

    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("known_for_department")
    val knownForDepartment: String?,

    @all:JsonProperty("name")
    val name: String?,

    @all:JsonProperty("original_name")
    val originalName: String?,

    @all:JsonProperty("popularity")
    val popularity: Double,

    @all:JsonProperty("profile_path")
    val profilePath: String?,

    @all:JsonProperty("roles")
    val roles: List<TvSeriesAggregateCastRole> = emptyList(),

    @all:JsonProperty("total_episode_count")
    val totalEpisodeCount: Int,

    @all:JsonProperty("order")
    val order: Int
) : TmdbModel